### DISCUSSION TITLE
hiedr2mosaic.py problem

### DISCUSSION THREAD

**Discussion by AndreaM92** [NeoGeographyToolkit/StereoPipeline#367] [Q&A]
Hello,

First of all, sorry if i am posting in the wrong section, i am not familiar with this and i thought Q&A was the right place.
My name is Andrea, and I am a PhD student who is trying to learn how to use the StereoPipeline software to create DEMs of Mars.

I went through the manual "asp_book.pdf," especially the chapter "Stereo Processing Examples" but every time i try to run the command hiedr2mosaic.py, i get the following lines:

(asp) bash-3.2$ hiedr2mosaic.py PSP_001513_1655_RED*.IMG
Traceback (most recent call last):
  File "/opt/anaconda3/envs/asp/bin/hiedr2mosaic.py", line 471, in <module>
    sys.exit(main())
  File "/opt/anaconda3/envs/asp/bin/hiedr2mosaic.py", line 405, in main
    if get_ccd(a) == options.match:
  File "/opt/anaconda3/envs/asp/bin/hiedr2mosaic.py", line 359, in get_ccd
    raise Exception('Could not extract the CCD number from: ' + filename)
Exception: Could not extract the CCD number from: PSP_001513_1655_RED*.IMG

As a result, i get nothing.

I also tried doing things manually, by first converting HiRISE images into .cub with "hi2isis," and everything worked well, but when i use "hical" to calibrate the images, i have no configuration file, and i do not know where i can get it.

If anyone have any suggestion on how to solve any of these two issues, please let me know.

**Comment by oleg-alexandrov** [comment]
There is a chance maybe the ISIS data is not installed or something about ISIS itself is not right. There is some documentation about that here and links from there: https://stereopipeline.readthedocs.io/en/latest/installation.html#quick-start-for-isis-users

I wonder what error you get when you run hical and how exactly you run it.

**Comment by AndreaM92** [comment]
I basically switch to isis and then run hical to open the program. What i do after is select the .cub file that needs calibration, then i specify the name and the folder to where i want the file to be saved.
I do not modify any of the other settings, as i do not really know if i have to modify them in the first place.
I attached a screenshot of the program with the error:

<img width="720" alt="Screenshot 2022-06-01 at 10 33 34" src="https://user-images.githubusercontent.com/106559058/171443248-9f6beeda-80c6-4eb9-9608-e8b0b5e5dad7.png">
 
It is my understanding that the configuration file is missing, but i do not know where to get one.

Thanks for the help,

Andrea

**Comment by rbeyer** [comment]
Yes, so this is definitely an ISIS thing.  So it looks like you followed the ISIS install directions to download the main ISIS "base" data, but if you are going to run these MRO programs, you also need to download the "mro" data, and you can read more about mission specific downloads [here](https://github.com/USGS-Astrogeology/ISIS3#mission-specific-data-downloads).  Unfortunately, make sure you have about 200 GB free and the time to download the MRO data.

**Comment by oleg-alexandrov** [comment]
You will also need to set ISISDATA per the doc. Your error shows that it is looking in the ISIS installation directory, not in the ISIS data directory.
